gnuradio-runtime/lib/block.cc

```diff
@@ -49,7 +49,8 @@ block::block(const std::string& name,
 {
     global_block_registry.register_primitive(alias(), this);
     message_port_register_in(d_system_port);
-    set_msg_handler(d_system_port, boost::bind(&block::system_handler, this, _1));
+    set_msg_handler(d_system_port,
+                    std::bind(&block::system_handler, this, std::placeholders::_1));
```

Comment:
Is there a style preference between `std::bind` and using lambdas?

```
set_msg_handler(d_system_port, [this]{ system_handler(); });
```

Reply:
I think the lambdas are much cleaner - I was given feedback to use that form when implementing this block:
https://github.com/gnuradio/gnuradio/blob/bfeef69cc9bdec8e721e07b444392708098cfb90/gr-blocks/lib/selector_impl.cc#L44

Reply:
Looking at the lambda version, it looks way cleaner and better understandable. There's just one catch. There a a lot of places where this needs to be updated.

Reply:
I did some checking on godbolt and benchmarking locally. Lambda seems to be about 3.5x faster than `bind` to call, too.

```c++
#include<functional>
void foo(std::function<void(int)> f)
{
  f(0);
}
```

```c++
#include<iostream>
#include<vector>
#include<functional>

void foo(std::function<void(int)> f);

class Foo {
 public:
  void f(int a){};
  void bar() {
    foo(std::bind(&Foo::f, this, std::placeholders::_1));
  }
  void baz() {
    foo([this](int a){
          this->f(a);
        });
  }
};

int main() {
  Foo f;
  for (int i = 0; i < 1000000000; i++) {
    f.bar();
    //f.baz();
  }
}
```

Reply:
So each call to the bound function is faster with lambdas? That's a very good reason to move to lambdas.

Reply:
Yeah. Ugh, I'm slow to wake up today, so my benchmark code doesn't actually isolate the call itself.

This should be a replacement for the second file:

```c++

class Foo {
 public:
  std::function<void(int)> cb;
  void f(int a){};
};

int main() {
  Foo f;

#if 1
  f.cb = std::bind(&Foo::f, &f, std::placeholders::_1);
#else
  f.cb = [&f](int a){
          f.f(a);
        };
#endif
  for (int i = 0; i < 1000000000; i++) {
    foo(f.cb);
  }
}
```

Still 3.5x cheaper with lambda, yes.

Reply:
Using lambdas has the additional advantage that we can use it with both boost::function and std::function